### PR TITLE
PolySwarm V2 Update

### PR DESCRIPTION
**Pull request by JavierBotella**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [X] Tests
- [X] Documentation

### CODE REVIEW

**Comment by mchasepan** [comment]
@JavierBotella I thought you said you'd also add the README.md into this PR? Can you please do so?

**Comment by JavierBotella** [comment]
> @JavierBotella I thought you said you'd also add the README.md into this PR? Can you please do so?

Yes, on it. We have spoken that in Slack already, sorry about it.

It's WIP PR for the devs to review in the meanwhile. 

Thanks you @mchasepan

**Comment by JavierBotella** [comment]
> @JavierBotella I thought you said you'd also add the README.md into this PR? Can you please do so?

Added! @mchasepan

**Review by Noy-Maimon — CHANGES_REQUESTED** [review]

File: Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm.py [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+        except Exception as err:
+            return_error('{ERROR_ENDPOINT}{err}'.
+                         format(ERROR_ENDPOINT=ERROR_ENDPOINT,
+                                err=err))
+
+        return self._get_results(title, total_scans,
+                                 positives, result.permalink,
+                                 param['hash'])
+
+    def url_reputation(self,
```
[the remark this answers is not in the document]
Comment (JavierBotella) [reply Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm.py:195]: done at https://github.com/demisto/content/pull/12424/commits/f1f2e757a9e6dfe964cebe58ce9cd3a996baaf0d

File: Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm.py [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+            results = self.polyswarm_api.search(EICAR_HASH)
+            for result in results:
+                if result.failed:
+                    return False
+        except Exception:
+            return False
+
+        return True
+
+    def file_reputation(self,
```
[the remark this answers is not in the document]
Comment (JavierBotella) [reply Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm.py:67]: done at https://github.com/demisto/content/pull/12424/commits/f1f2e757a9e6dfe964cebe58ce9cd3a996baaf0d

[the remark this answers is not in the document]
Comment (JavierBotella) [reply Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm.py]: and this one https://github.com/demisto/content/pull/12424/commits/7321b5af25f0adb707812346a82fa6b520f2f06c

File: Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm.py [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+                     positives: int,
+                     permalink: str,
+                     artifact: str) -> dict:
+        contxt = makehash()
+
+        contxt['Scan_UUID'] = artifact
+        contxt['Total'] = str(total_scans)
+        contxt['Positives'] = str(positives)
+        contxt['Permalink'] = permalink
+        contxt['Artifact'] = artifact
```
[the remark this answers is not in the document]
Comment (JavierBotella) [reply Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm.py]: Dbotscore with PolyScore implemented at https://github.com/demisto/content/pull/12424/commits/6e31daa1d0345011bc799aa98bfcc0bbf8d0b144

File: Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml [hunk 4]
```diff
@@ -0,0 +1,398 @@
...
+    - contextPath: PolySwarm.Positives
+      description: The total number of positives found.
+      type: Number
+    - contextPath: PolySwarm.Scan_UUID
+      description: The PolySwarm scan UUID.
+      type: String
+    - contextPath: PolySwarm.Artifact/
+      description: The artifact queried.
+      type: String
+  outputs:
```
Comment (Noy-Maimon) [inline Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml:163]:
```suggestion
```

File: Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml [hunk 5]
```diff
@@ -0,0 +1,398 @@
...
+    - contextPath: PolySwarm.Permalink
+      description: PolySwarm permalink results.
+      type: String
+    - contextPath: PolySwarm.Positives
+      description: The total number of positives found.
+      type: Number
+    - contextPath: PolySwarm.Scan_UUID
+      description: The PolySwarm scan UUID.
+      type: String
+    - contextPath: PolySwarm.Artifact/
```
Comment (Noy-Maimon) [inline Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml:160]:
```suggestion
    - contextPath: PolySwarm.Artifact
```

File: Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml [hunk 6]
```diff
@@ -0,0 +1,398 @@
...
+  name: api_key
+  required: true
+  type: 4
+- defaultvalue: https://api.polyswarm.network/v2
+  display: The base URL to connect to
+  name: base_url
+  required: true
+  type: 0
+- defaultvalue: default
+  display: PolySwarm community
```
Comment (Noy-Maimon) [inline Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml:16]:
```suggestion
  display: PolySwarm V2
```

File: Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml [hunk 7]
```diff
@@ -0,0 +1,398 @@
...
+  name: polyswarm_community
```
Comment (Noy-Maimon) [inline Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml:17]:
```suggestion
  name: Polyswarm V2
```

File: Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml [hunk 8]
```diff
@@ -0,0 +1,398 @@
...
+    - contextPath: PolySwarm.Scan_UUID
+      description: The PolySwarm scan UUID.
+      type: String
+    - contextPath: PolySwarm.Artifact
+      description: The artifact queried.
+      type: String
+  - arguments:
+    - default: true
+      description: The IP address for which to retrieve the reputation information.
+      isArray: false
```
Comment (Noy-Maimon) [inline Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml:88]:
```suggestion
      isArray: true
```

File: Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml [hunk 9]
```diff
@@ -0,0 +1,398 @@
...
+    - contextPath: URL.MalwareFamily
+      description: The malware family associated with the url.
+      type: String
+    - contextPath: URL.Tags
+      description: Tags that are associated with the url.
+      type: String
+  - arguments:
+    - default: true
+      description: Domain for which to retrieve the reputation information.
+      isArray: false
```
Comment (Noy-Maimon) [inline Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml:191]:
```suggestion
      isArray: true
```

File: Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml [hunk 10]
```diff
@@ -0,0 +1,398 @@
...
+    - contextPath: Domain.MalwareFamily
+      description: The malware family associated with the domain.
+      type: String
+    - contextPath: Domain.Tags
+      description: Tags that are associated with the domain.
+      type: String
+  - arguments:
+    - default: true
+      description: The URL to scan.
+      isArray: false
```
Comment (Noy-Maimon) [inline Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml:242]:
```suggestion
      isArray: true
```

File: Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml [hunk 11]
```diff
@@ -0,0 +1,398 @@
...
+  - arguments:
+    - default: false
+      description: 'The value of the file hash for which to retrieve the reputation information. The hash type can be: "SHA256", "SHA1", or "MD5".'
+      isArray: false
+      name: hash
+      required: false
+      secret: false
+    - default: true
+      description: 'The type of the file hash for which to retrieve the reputation information. The hash type can be: "SHA256", "SHA1", or "MD5".'
+      isArray: false
```
Comment (Noy-Maimon) [inline Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml:61]:
```suggestion
      isArray: true
```

**Review by Noy-Maimon — CHANGES_REQUESTED** [review]

File: Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml [hunk 12]
```diff
@@ -0,0 +1,397 @@
...
+  required: true
+  type: 4
+- defaultvalue: https://api.polyswarm.network/v2
+  display: The base URL to connect to
+  name: base_url
+  required: true
+  type: 0
+- defaultvalue: default
+  display: PolySwarm_v2 Community
+  name: PolySwarm_v2 Community
```
Comment (Noy-Maimon) [inline Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml:17]:
```suggestion
  name: PolySwarm_v2
```

**Review by Noy-Maimon — CHANGES_REQUESTED** [review]

**Review by Noy-Maimon — APPROVED** [review]

File: Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml [hunk 13]
```diff
@@ -0,0 +1,397 @@
...
+- defaultvalue: https://api.polyswarm.network/v2
+  display: The base URL to connect to
+  name: base_url
+  required: true
+  type: 0
+- defaultvalue: default
+  display: PolySwarm_v2 Community
+  name: PolySwarm_v2
+  required: true
+  type: 0
```
Comment (Noy-Maimon) [inline Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml:19]:
```suggestion
  type: 0
  description: the segment of PolySwarm's marketplace to query on.
```

**Review by Noy-Maimon — CHANGES_REQUESTED** [review]

**Comment by Noy-Maimon** [comment]
@roysagi Hi, we added a _description.md file to the PR (all of the other filles were already approved). Can you please have a look at it? Thanks

File: Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml [hunk 14]
```diff
@@ -0,0 +1,398 @@
...
+  name: api_key
+  required: true
+  type: 4
+- defaultvalue: https://api.polyswarm.network/v2
+  display: The base URL to connect to
+  name: base_url
+  required: true
+  type: 0
+- defaultvalue: default
+  display: PolySwarm_v2 Community
```
Comment (Noy-Maimon) [inline Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml:16]:
```suggestion
  display: PolySwarm v2 Community
```

Comment (Noy-Maimon) [reply Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml:16]: This is needed for passing our validations before merging

**Review by Noy-Maimon — CHANGES_REQUESTED** [review]

File: Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2_test.py [hunk 15]
```diff
@@ -0,0 +1,1025 @@
+
+import demistomock as demisto
+
+from PolySwarm import PolyswarmConnector
```
Comment (Noy-Maimon) [inline Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2_test.py:4]:
```suggestion
from PolySwarm_v2 import PolyswarmConnector
```

**Review by Noy-Maimon — CHANGES_REQUESTED** [review]

File: Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.py [hunk 16]
```diff
@@ -0,0 +1,356 @@
...
+        return True
+
+    def get_score(self,
+                  polyscore) -> int:
+
+        if float(polyscore) < 0.2:
+            return Common.DBotScore.GOOD
+        elif 0.2 <= float(polyscore) < 0.7:
+            return Common.DBotScore.SUSPICIOUS
+        elif float(polyscore) >= 0.7:
```
Comment (Noy-Maimon) [inline Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.py:77]:
```suggestion
        else:  # polyscore is >= 0.7
```

File: Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.py [hunk 17]
```diff
@@ -0,0 +1,356 @@
...
+                   'Artifact': artifact}
+
+        readable_output = tableToMarkdown(title,
+                                          results)
+        return CommandResults(
+            readable_output=readable_output,
+            outputs_prefix=f'PolySwarm.{object_name}',
+            outputs_key_field='Scan_UUID',
+            outputs=results,
+            indicator=indicator,
```
Comment (Noy-Maimon) [inline Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.py:54]:
```suggestion
            indicator=indicator, # type: ignore
```

File: Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.py [hunk 18]
```diff
@@ -0,0 +1,356 @@
...
+
+        if float(polyscore) < 0.2:
+            return Common.DBotScore.GOOD
+        elif 0.2 <= float(polyscore) < 0.7:
+            return Common.DBotScore.SUSPICIOUS
+        elif float(polyscore) >= 0.7:
+            return Common.DBotScore.BAD
+
+    def return_hash_results(self,
+                            results: dict,
```
Comment (Noy-Maimon) [inline Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.py:81]:
```suggestion
                            results: list,
```

File: Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.py [hunk 19]
```diff
@@ -0,0 +1,356 @@
...
+                                          integration_name='PolySwarm',
+                                          score=self.get_score(polyscore))
+
+            indicator = None
+            if artifact_type == 'ip':
+                indicator = Common.IP(ip=artifact,
+                                      dbot_score=dbot_score)
+            elif artifact_type == 'url':
+                indicator = Common.URL(url=artifact,
+                                       dbot_score=dbot_score)
```
Comment (Noy-Maimon) [inline Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.py:293]:
```suggestion
                                       dbot_score=dbot_score)  # type: ignore
```

File: Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.py [hunk 20]
```diff
@@ -0,0 +1,356 @@
...
+            elif artifact_type == 'domain':
+                indicator = Common.Domain(domain=artifact,
+                                          dbot_score=dbot_score)
```
Comment (Noy-Maimon) [inline Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.py:296]:
```suggestion
                                          dbot_score=dbot_score)  # type: ignore
```

**Review by Noy-Maimon — CHANGES_REQUESTED** [review]

File: Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.py [hunk 21]
```diff
@@ -0,0 +1,356 @@
...
+                   'Artifact': artifact}
+
+        readable_output = tableToMarkdown(title,
+                                          results)
+        return CommandResults(
+            readable_output=readable_output,
+            outputs_prefix=f'PolySwarm.{object_name}',
+            outputs_key_field='Scan_UUID',
+            outputs=results,
+            indicator=indicator, # type: ignore
```
Comment (Noy-Maimon) [inline Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.py:54]:
```suggestion
            indicator=indicator,  # type: ignore
```

**Review by Noy-Maimon — CHANGES_REQUESTED** [review]

File: Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml [hunk 22]
```diff
@@ -0,0 +1,398 @@
...
+  display: The base URL to connect to
+  name: base_url
+  required: true
+  type: 0
+- defaultvalue: default
+  display: PolySwarm v2 Community
+  name: PolySwarm_v2
+  required: true
+  type: 0
+  description: the segment of PolySwarm's marketplace to query on.
```
Comment (Noy-Maimon) [inline Packs/PolySwarm/Integrations/PolySwarm_v2/PolySwarm_v2.yml:20]:
```suggestion
  additionalinfo: the segment of PolySwarm's marketplace to query on.
```

**Review by Noy-Maimon — CHANGES_REQUESTED** [review]

**Comment by Noy-Maimon** [comment]
@amshamah419  Hi, we added a _description.md file to the PR (all of the other filles were already approved). Can you please have a look at it? Thanks

**Review by Noy-Maimon — APPROVED** [review]

**Pull request merged**